src/Marten/Events/CodeGeneration/AggregateEventProcessingFrame.cs

```diff
@@ -78,10 +79,15 @@ public override void GenerateCode(GeneratedMethod method, ISourceWriter writer)
 
             if (Apply != null)
             {
-                if (AggregateType.GetConstructors().Any(x => !x.GetParameters().Any()))
+                var defaultConstructor = AggregateType.GetConstructors(BindingFlags.Public | BindingFlags.NonPublic).SingleOrDefault(x => !x.GetParameters().Any());
```

Comment:
I still think this should only happen if there's no specific CreationFrame, otherwise it's stomping on what's already been created.